crypto/x509/x_all.c

```diff
@@ -153,14 +153,14 @@ int X509_verify(X509 *a, EVP_PKEY *r)
 #endif
 
     if (X509_ALGOR_cmp(&a->sig_alg, &a->cert_info.signature))
-        return 0;
+        return -1;
```

Comment:
I think when you change the return value to -1 here you also need to push an error to the error queue

Reply:
Good point - done.

On this occasion I've also replaced a handful of (wrong) `ASN1_R_` error code prefixes in `x_all.c` by `X509_R_`.